controllers/useraccount/useraccount_controller_test.go

```diff
@@ -477,6 +478,159 @@ func TestReconcile(t *testing.T) {
+				Namespace: UserSettingNS,
+			},
+		}
+		role := &v1.Role{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName + "random",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+					ConsoleUserSettingsUID:        string(userUID),
+				},
+			},
+		}
+		rb := &v1.RoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName,
+				Namespace: UserSettingNS,
+			},
+		}
+		noiseCm := &corev1.ConfigMap{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "ConfigMap",
+				APIVersion: "v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "noise-object",
+				Namespace: UserSettingNS,
+			},
+		}
+		noiseRole := &v1.Role{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "Role",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      resourceName + "noise-role",
+				Namespace: UserSettingNS,
+				Labels: map[string]string{
+					ConsoleUserSettingsIdentifier: "true",
+				},
+			},
+		}
+		noiseRb := &v1.RoleBinding{
+			TypeMeta: metav1.TypeMeta{
+				Kind:       "RoleBinding",
+				APIVersion: "rbac.authorization.k8s.io/v1",
+			},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "noise-rb",
+				Namespace: UserSettingNS,
+			},
+		}
+		r, req, cl, _ := prepareReconcile(t, username, cfg, userAcc, preexistingUser, preexistingIdentity, configMap, role, rb, noiseCm, noiseRole, noiseRb)
+
+		t.Run("first reconcile deletes identity", func(t *testing.T) {
+			// given
+			// Set the deletionTimestamp
+			now := metav1.NewTime(time.Now())
+			userAcc.DeletionTimestamp = &now
+			err = r.Client.Update(context.TODO(), userAcc)
+			require.NoError(t, err)
+
+			// when
+			res, err := r.Reconcile(context.TODO(), req)
+
+			// then
+			assert.Equal(t, reconcile.Result{}, res)
+			require.NoError(t, err)
+
+			useraccount.AssertThatUserAccount(t, req.Name, cl).
+				HasConditions(notReady("Terminating", "deleting user/identity"))
+
+			// Check that the associated identity has been deleted
+			// when reconciling the useraccount with a deletion timestamp
+			assertIdentityNotFound(t, r, userAcc, config.Auth().Idp())
+			useraccount.AssertThatUserAccount(t, userAcc.Name, cl).HasConditions(terminating())
+
+			t.Run("second reconcile deletes user and its resources - configmap, role and rolebinding", func(t *testing.T) {
+				// when
+				res, err = r.Reconcile(context.TODO(), req)
+
+				// then
+				assert.Equal(t, reconcile.Result{}, res)
+				require.NoError(t, err)
+
+				useraccount.AssertThatUserAccount(t, req.Name, cl).
+					HasConditions(notReady("Terminating", "deleting user/identity"))
+
+				// Check that the associated configmap has been deleted
+				retrievedConfigMap := &corev1.ConfigMap{}
+				err := r.Client.Get(context.TODO(), types.NamespacedName{Name: resourceName, Namespace: UserSettingNS}, retrievedConfigMap)
+				require.Error(t, err)
+				assert.True(t, apierros.IsNotFound(err))
```

Comment:
this (and the following lines) can be replaced with the usage of 
```go
AssertObjectNotFound(t, cl, UserSettingNS, resourceName, &corev1.ConfigMap{})
```

Reply:
done in [7b7bd25](https://github.com/codeready-toolchain/member-operator/pull/568/commits/7b7bd259bb5cb92d555a5043585a0d3c06875e8b)